strapi/strapi
# [v5] use fetch api for strapi.fetch

### What does it do?

- Removes dependency for node-fetch throughout strapi
- Replaces strapi.fetch object with a native Fetch wrapper instead of a node-fetch wrapper
- refactor any node-fetch <-> fetch usage
- replaces the https-global-proxy with an undici ProxyAgent

Note: the plan was and is to remove all third-party modules and exclusively use the Node core Fetch API for everything related to requests, but since fetch is currently in a state between experimental and stable (that is, Node has removed the experimental flag but not yet declared it stable) we will rely on Undici to fill any gaps since its API is identical. If we discover the fetch API to be too unstable for the v5 release, we will rely fully on undici until it can be removed at a later date.

### Why is it needed?

To have a standardized way of performing requests on the backend in the monorepo instead of a wide variety of libraries being used

### How to test it?

- strapi.fetch should work pretty much as before (except now it wraps fetch instead of node-fetch)
- telemetry should work (had to update some calls that now use fetch directly instead of node-fetch)

### Related issue(s)/PR(s)

Roadmap in upcoming PRs:

For every backend request:

- where strapi object is available, use strapi.fetch
- where strapi object is not available, use fetch API directly (replace http/https, request, axios, etc)
